jamshidbek/db.sql

```diff
@@ -0,0 +1,187 @@
+('Emily', 'Wilson', '2002-02-02', 3.7),
+('David', 'Taylor', '2001-09-10', 3.1),
+('Grace', 'Miller', '1999-12-01', 3.8),
+('Henry', 'Moore', '2000-05-15', 3.4),
+('Isabel', 'Anderson', '1998-08-20', 3.9);
+
+select *from studentt;
+
+
+--  2
+
+ create table courses(
+course_id serial primary key,
+course_name varchar(100),
+credit_hours INTEGER
+);
+
+INSERT INTO courses (course_name, credit_hours)
+VALUES 
+    ('Mathematics', 3),
+    ('History', 4),
+    ('Physics', 3),
+    ('Computer Science', 3),
+    ('Chemistry', 4),
+    ('Biology', 3),
+    ('English Literature', 2),
+    ('Economics', 3),
+    ('Psychology', 3),
+    ('Sociology', 3);
+select * from courses;
+
+create table enrollments(
+	enrollment_id serial primary key, 
+	student_id int,
+	foreign key (student_id) REFERENCES studentt(student_id),
+	course_id int,
+	foreign key (course_id) REFERENCES courses(course_id),
+	grade int
+	);
+
+INSERT INTO enrollments (student_id, course_id, grade)
+VALUES (1, 1, 90),  -- Student 1, Mathematics, grade 90
+       (1, 2, 85),  -- Student 1, History, grade 85
+       (1, 3, 88),  -- Student 1, Physics, grade 88
+       (1, 4, 91),  -- Student 1, Computer Science, grade 91
+       (2, 1, 78),  -- Student 2, Mathematics, grade 78
+       (2, 2, 84),  -- Student 2, History, grade 84
+       (2, 5, 92),  -- Student 2, Chemistry, grade 92
+       (2, 6, 76),  -- Student 2, Biology, grade 76
+       (3, 3, 87),  -- Student 3, Physics, grade 87
+       (3, 4, 93),  -- Student 3, Computer Science, grade 93
+       (3, 5, 89),  -- Student 3, Chemistry, grade 89
+       (3, 7, 80),  -- Student 3, English Literature, grade 80
+       (4, 6, 72),  -- Student 4, Biology, grade 72
+       (4, 8, 88),  -- Student 4, Economics, grade 88
+       (4, 9, 85),  -- Student 4, Psychology, grade 85
+       (5, 2, 90),  -- Student 5, History, grade 90
+       (5, 3, 81),  -- Student 5, Physics, grade 81
+       (5, 4, 92),  -- Student 5, Computer Science, grade 92
+       (6, 1, 85),  -- Student 6, Mathematics, grade 85
+       (6, 5, 80),  -- Student 6, Chemistry, grade 80
+       (6, 6, 83),  -- Student 6, Biology, grade 83
+       (7, 2, 95),  -- Student 7, History, grade 95
+       (7, 7, 77),  -- Student 7, English Literature, grade 77
+       (7, 8, 70),  -- Student 7, Economics, grade 70
+       (8, 9, 91),  -- Student 8, Psychology, grade 91
+       (8, 10, 84), -- Student 8, Sociology, grade 84
+       (9, 1, 79),  -- Student 9, Mathematics, grade 79
+       (9, 2, 80),  -- Student 9, History, grade 80
+       (10, 3, 88); -- Student 10, Physics, grade 88
+
+select *from enrollments;
+
+--  3.1
+select first_name,last_name,date_of_birth
+from studentt;
+
+select s.first_name,s.first_name,s.date_of_birth
+from studentt s 
+join enrollments e on s.student_id = e.student_id
+join courses c on e.course_id = c.course_id
+where course_name = 'Mathematics';
+
+SELECT first_name, last_name, date_of_birth, gpa
+FROM studentt
+WHERE gpa < 4;
```

Comment:
As I mentioned, you should not have a GPA; instead, calculate grades using the enrollment table.